modules/catalog/src/main/java/org/apache/ignite/internal/catalog/storage/UpdateLogImpl.java

```diff
@@ -167,7 +169,7 @@ private void restoreStateFromVault(OnUpdateHandler handler) {
 
             VersionedUpdate update = fromBytes(entry.value());
 
-            handler.handle(update);
+            handler.handle(update, metastore.appliedRevision());
```

Comment:
This has nothing to do with the fix. Please do it in a separate issue.

Reply:
But the catalog creates events that are descendants of `org.apache.ignite.internal.manager.EventParameters` that require `causalityToken`, the catalog now passes the correct value.

Reply:
It's only required now because you added it, but ok, this change is not as big as the rest